src/qtt/measurements/scans.py

```diff
@@ -881,7 +881,18 @@ def _update_step_size_from_length(data, length):
                         data['step'] = (data['end'] - data['start']) / (length-1)
 
         def _calculate_sweepvalues(param, sweep_data) -> SweepFixedValues:
-            """ From parameter and sweep data create a SweepFixedValues object """
+            """ From parameter and sweep data create a SweepFixedValues object. When (end - start) / step is not an
+            integer value, the end-value will be adjusted (lowered) until it is.
+             """
+            if sweep_data['step'] is not None:
```

Comment:
Is the `if` needed? The `sweep_data` can be input at high level without a `step`, but at this stage it should always have a value right? If it is None, then the construction of SweepFixedValues will fail.

Reply:
You are right, I will delete this test